Remove duplicate JAXP dependency saxon

billy-gin is dependent on two implementations of JAXP:
  * `xalan` transitively through fop
  * `saxon` directly
  
This was a problem with our application when processing a XML file not related with Billy, for ehcache configuration. 

On one machine Java was loading `xalan` and working fine, on another machine it was loading `saxon` and failing with:
```
	Caused by: java.lang.IllegalArgumentException: Unknown type of result: class javax.xml.transform.dom.DOMResult
		at net.sf.saxon.event.SerializerFactory.getReceiver(SerializerFactory.java:154)
		at net.sf.saxon.IdentityTransformerHandler.startDocument(IdentityTransformerHandler.java:94)
		at com.sun.xml.bind.v2.runtime.unmarshaller.DomLoader$State.<init>(DomLoader.java:53)
		at com.sun.xml.bind.v2.runtime.unmarshaller.DomLoader.startElement(DomLoader.java:88)
		at com.sun.xml.bind.v2.runtime.unmarshaller.ProxyLoader.startElement(ProxyLoader.java:30)
		at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:545)
		at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524)
		at com.sun.xml.bind.v2.runtime.unmarshaller.InterningXmlVisitor.startElement(InterningXmlVisitor.java:45)
		at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.startElement(SAXConnector.java:137)
		at com.sun.xml.bind.unmarshaller.DOMScanner.visit(DOMScanner.java:214)
		at com.sun.xml.bind.unmarshaller.DOMScanner.visit(DOMScanner.java:251)
		at com.sun.xml.bind.unmarshaller.DOMScanner.visit(DOMScanner.java:220)
		at com.sun.xml.bind.unmarshaller.DOMScanner.visit(DOMScanner.java:251)
		at com.sun.xml.bind.unmarshaller.DOMScanner.visit(DOMScanner.java:220)
		at com.sun.xml.bind.unmarshaller.DOMScanner.scan(DOMScanner.java:97)
		at com.sun.xml.bind.unmarshaller.DOMScanner.scan(DOMScanner.java:80)
		at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:341)
		at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:318)
		at org.ehcache.xml.ConfigurationParser.documentToConfig(ConfigurationParser.java:253)
		at org.ehcache.xml.XmlConfiguration.<init>(XmlConfiguration.java:116)
```

The difference between machines is because on Tomcat, the classpath is an [unordered list of files](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#list--) of the lib folder. The order is filesystem dependent. 

Our workaround is to define a system property to specify the JAXP implementation:
```
-Djavax.xml.transform.TransformerFactory=org.apache.xalan.processor.TransformerFactoryImpl
```

--- a/billy-gin/pom.xml
+++ b/billy-gin/pom.xml
@@ -74,11 +74,6 @@
 			<artifactId>fop</artifactId>
 			<version>1.0</version>
 		</dependency>
-		<dependency>
-			<groupId>net.sf.saxon</groupId>
-			<artifactId>saxon</artifactId>
-			<version>8.7</version>
-		</dependency>
 
 		<!-- QRCode -->
 		<dependency>

--- a/billy-gin/src/main/java/com/premiumminds/billy/gin/services/impl/pdf/FOPPDFTransformer.java
+++ b/billy-gin/src/main/java/com/premiumminds/billy/gin/services/impl/pdf/FOPPDFTransformer.java
@@ -47,9 +47,9 @@ import javax.xml.transform.Source;
 import javax.xml.transform.Transformer;
 import javax.xml.transform.TransformerConfigurationException;
 import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
 import javax.xml.transform.sax.SAXResult;
 import javax.xml.transform.stream.StreamSource;
-import net.sf.saxon.TransformerFactoryImpl;
 import org.apache.commons.lang3.StringEscapeUtils;
 import org.apache.fop.apps.FOPException;
 import org.apache.fop.apps.FOUserAgent;
@@ -65,14 +65,14 @@ public abstract class FOPPDFTransformer {
     private static final String QR_CODE_PATH = "qrCodePath";
     private static final String QR_CODE = "qrCode";
 
-    private final TransformerFactoryImpl transformerFactory;
+    private final TransformerFactory transformerFactory;
 
-    public FOPPDFTransformer(TransformerFactoryImpl transformerFactory) {
+    public FOPPDFTransformer(TransformerFactory transformerFactory) {
         this.transformerFactory = transformerFactory;
     }
 
     public FOPPDFTransformer() {
-        this(new TransformerFactoryImpl());
+        this(TransformerFactory.newInstance());
     }
 
     private Source mapParamsToSource(ParamsTree<String, String> documentParams) {

